packages/@guidesmiths/cuckoojs-cli/src/cli.ts

```diff
@@ -19,13 +20,30 @@ const init = (): void => {
 		.usage('<command> [options]');
 
 	cuckoo
-		.command('new <name>')
+		.command('new <name> [options...]')
 		.alias('n')
 		.description('Generate Nest application with basic tooling.')
 		.action(async (name: string, _options: any) => {
 			await new NewCommand(name).execute();
 		});
 
+	// Absolutely provisional until this is refined with the team to see how to merge or work with both project types
+	cuckoo
```

Comment:
Suggestion from the Chapter: could we use this https://github.com/tj/commander.js/blob/master/examples/nestedCommands.js for subcommands? And maybe have:

- `nest`
  - `new`
  - `generate`
- `lambda`
  - `new`

Reply:
```js
	const nest = cuckoo
		.command('nest')
		.description('Generate Nest components');

	nest
		.command('new <name>')
		.alias('n')
		.description('Generate Nest application with basic tooling.')
		.action(async (name: string, _options: any) => {
			await new NewCommand(name).execute();
		});

	nest
		.command('generate <schematic> <name> [options...]')
		.alias('g')
		.description('Generate a Nest element.')
		.action(async (schematic: string, name: string, options: string[]) => {
			await new GenerateCommand(schematic, name, options).execute();
		});

	const lambda = cuckoo
		.command('lambda')
		.description('Generate lambda templates');

	lambda
		.command('new <name>')
		.alias('n')
		.description('Generate a lambda template using Awilix')
		.action(async (_name: string, _options: any) => {
			console.log('Hello');
		});
```